doc/tutorial/network_ring.rst

```diff
@@ -0,0 +1,337 @@
+
+Step **(8)** creates a :class:`arbor.connection` between this cell and the previous (the ``gid`` of the previous cell is ``(gid-1)%self.ncells``), with a weight of 0.1 μS and a delay of 5 ms. The two arguments to :class:`arbor.cell_member` refer to the cell ``gid`` (first argument) and the index of the synapse (second argument). Only one synapse was defined (step **(4)**), so the index is always 0. :func:`arbor.cable_cell.num_targets` and :func:`arbor.cable_cell.num_sources` must be set to 1: each cell has one connection coming in and one going out.
+
+In step **(9)** an :class:`arbor.event_generator` is created on the 0th cell. The :class:`arbor.explicit_schedule` in instantiated with a list of times with unit ms, so a schedule with a period of a millisecond is created.
+
+.. code-block:: python
+
+   # (5) Create a recipe that generates a network of connected cells.
+   class ring_recipe (arbor.recipe):
+      def __init__(self, ncells):
+         # The base C++ class constructor must be called first, to ensure that
+         # all memory in the C++ class is initialized correctly.
+         arbor.recipe.__init__(self)
+         self.ncells = ncells
+         self.props = arbor.neuron_cable_properties()
+         self.cat = arbor.default_catalogue()
+         self.props.register(self.cat)
+
+      # (6) The num_cells method that returns the total number of cells in the model
+      # must be implemented.
+      def num_cells(self):
+         return self.ncells
+
+      # (7) The cell_description method returns a cell
+      def cell_description(self, gid):
+         return make_cable_cell(gid)
+
+      # The kind method returns the type of cell with gid.
+      # Note: this must agree with the type returned by cell_description.
+      def cell_kind(self, gid):
+         return arbor.cell_kind.cable
+
+      # (8) Make a ring network
+      def connections_on(self, gid):
+         src = (gid-1)%self.ncells
+         w = 0.01
+         d = 5
+         return [arbor.connection(arbor.cell_member(src,0), arbor.cell_member(gid,0), w, d)]
+
+      def num_targets(self, gid):
+         return 1
+
+      def num_sources(self, gid):
+         return 1
+
+      # (9) Attach a generator to the first cell in the ring.
+      def event_generators(self, gid):
+         if gid==0:
+               sched = arbor.explicit_schedule([1])
+               return [arbor.event_generator(arbor.cell_member(0,0), 0.1, sched)]
+         return []
+
+      def probes(self, gid):
+         return [arbor.cable_probe_membrane_voltage('"root"')]
+
+      def global_properties(self, kind):
+         return self.props
+
+The execution context
+*********************
+
+The execution context contains all system-specific information needed by the simulation: it contains the
```

Comment:
An execution context describes the hardware resources on which the simulation will run. Then provide a link to the appropriate documentation.

It also does not hold GPU attributes.
It `It contains the thread pool used to parallelise work on the local CPU, and optionally describes GPU and MPI resources to use`.

Reply:
This is mostly copied from `single_cell_detailed_recipe.rst:L287` so I've made the same modification there.